File: c++/classTestExtend.cpp
#include <iostream>

using namespace std;


class Person
{
	string name;
	int age;

public:
	Person(string name, int age){
		this->name=name;
		this->age=age;
	}
	void say(){
		cout << this->name << endl;
		cout << this->age << endl;
	}
};

class Student: Person{
	int grade;

public:
	Student(string name, int age, int grade):Person(name, age){
		this->grade=grade;
	}
	void say(){
		Person::say();
		cout << this->grade << endl;
	}
};

void test(){
	Student s = Student("Stanley", 20, 8);
	s.say();
}

int main(){
	test();
	return 0;
}

File: swift/classTestExtend.swift
class Person{
    let name:String
    let age:Int
    
    init(name:String, age:Int){
        self.name = name
        self.age = age
    }
    
    func say(){
        print(self.name, self.age)
    }
}

class Student:Person{
    let glade:Int
    
    init(name:String, age:Int, glade:Int){
        self.glade = glade
        super.init(name:name, age:age)
    }
    
    override func say(){
        print(self.name, self.age, self.glade)
    }
}

func test(){
    let a = Student(name:"Stanley", age:20, glade: 8)
    a.say()
}